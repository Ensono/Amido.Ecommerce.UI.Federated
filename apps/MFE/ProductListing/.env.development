ASSETS_PATH=//localhost:3005
GENERATE_SOURCEMAP=true
PORT=3005
REMOTE_URLS={}
LOG_LEVEL=info