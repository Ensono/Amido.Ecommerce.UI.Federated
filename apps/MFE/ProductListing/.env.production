ASSETS_PATH=//localhost:3005
GENERATE_SOURCEMAP=false
PORT=3005
REMOTE_URLS={}
LOG_LEVEL=warn
