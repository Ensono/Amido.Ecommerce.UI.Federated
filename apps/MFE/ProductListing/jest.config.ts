export default {
  roots: ['<rootDir>/src'],
  collectCoverageFrom: ['src/**/*.{js,jsx,ts,tsx}', '!src/**/*.d.ts', '!src/remote-entry/**'],
  setupFiles: ['react-app-polyfill/jsdom'],
  setupFilesAfterEnv: ['<rootDir>/src/setupTests.ts'],
  testMatch: ['<rootDir>/src/**/__tests__/**/*.{js,jsx,ts,tsx}', '<rootDir>/src/**/*.{spec,test}.{js,jsx,ts,tsx}'],
  testEnvironment: 'jsdom',
  transform: {
    '^.+\\.(js|jsx|mjs|cjs|ts|tsx)$': '../../../config/jest/babelTransform.js',
    '^.+\\.css$': '../../../config/jest/cssTransform.js',
    '^(?!.*\\.(js|jsx|mjs|cjs|ts|tsx|css|json)$)': '../../../config/jest/fileTransform.js',
  },
  transformIgnorePatterns: [
    '[/\\\\]node_modules[/\\\\].+\\.(js|jsx|mjs|cjs|ts|tsx)$',
    '^.+\\.module\\.(css|sass|scss)$',
  ],
  modulePaths: [],
  moduleNameMapper: {
    '^react-native$': 'react-native-web',
    '^.+\\.module\\.(css|sass|scss)$': 'identity-obj-proxy',
  },
  moduleFileExtensions: ['web.js', 'js', 'web.ts', 'ts', 'web.tsx', 'tsx', 'json', 'web.jsx', 'jsx', 'node'],
  watchPlugins: ['jest-watch-typeahead/filename', 'jest-watch-typeahead/testname'],
  resetMocks: true,
  coverageReporters: ['json-summary', 'text'],
  coverageThreshold: {
    global: {
      lines: 0,
      statements: 0,
      functions: 0,
      branches: 0,
      branchesTrue: 0,
    },
  },
}
