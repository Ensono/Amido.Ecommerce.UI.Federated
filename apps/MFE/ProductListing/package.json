{
  "name": "@batman/product-listing",
  "version": "0.1.1",
  "private": true,
  "scripts": {
    "dev": "nodemon ../../../scripts/dev.js --config ../../../nodemon.config.json",
    "build": "node ../../../scripts/build.js",
    "serve": " node build/server.js",
    "watch": "nodemon --exec \"npm run build && npm run serve\" --config ../../../nodemon.config.json",
    "test": "node ../../../scripts/test.js --coverage --watchAll=false",
    "test:watch": "node ../../../scripts/test.js --watch",
    "jest-ratchet_DISABLED": "jest-coverage-ratchet",
    "lint": "eslint 'src/**/*.{jsx,js,tsx,ts}' --fix --ignore-path '../../../.eslintignore'",
    "prettier": "prettier 'src/**/*.{jsx,js,tsx,ts}' --write",
    "docs": "typedoc --out ../../../docs/product-listing src/*.*",
    "tsc": "tsc"
  },
  "dependencies": {
    "@batman/core-logger": "*",
    "@batman/federate-component": "*",
    "@batman/middlewares": "*",
    "@batman/remote-urls": "*"
  },
  "devDependencies": {
    "@batman/eslint-config": "1.0.0"
  },
  "browserslist": {
    "production": [
      ">0.2%",
      "not dead",
      "not op_mini all"
    ],
    "development": [
      "last 1 chrome version",
      "last 1 firefox version",
      "last 1 safari version"
    ]
  },
  "babel": {
    "presets": [
      "react-app"
    ],
    "plugins": [
      [
        "@babel/plugin-transform-react-jsx",
        {
          "runtime": "automatic"
        }
      ]
    ]
  }
}
