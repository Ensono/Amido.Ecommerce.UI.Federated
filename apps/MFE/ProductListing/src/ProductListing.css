.product-list {
  display: flex;
  justify-content: center;
}

.product-image {
  width: 200px;
}
