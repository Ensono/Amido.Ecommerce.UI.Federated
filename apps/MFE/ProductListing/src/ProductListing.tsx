/* eslint-disable jsx-a11y/no-static-element-interactions */
/* eslint-disable jsx-a11y/click-events-have-key-events */
import React from 'react'

const ProductListing: React.FC = () => {
  return (
    <div>
      <h2>Product Listing!!!</h2>
    </div>
  )
}

export default ProductListing
