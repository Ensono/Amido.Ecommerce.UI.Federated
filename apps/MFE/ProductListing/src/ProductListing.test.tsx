import {MemoryRouter} from 'react-router-dom'

import {render, waitFor} from '@testing-library/react'

import ProductListing from './ProductListing'

describe('ProductListing', () => {
  it('renders as expected', async () => {
    const {container} = render(
      <MemoryRouter>
        <ProductListing />
      </MemoryRouter>,
    )
    await waitFor(() => {
      expect(container).toMatchSnapshot()
    })
  })
})
