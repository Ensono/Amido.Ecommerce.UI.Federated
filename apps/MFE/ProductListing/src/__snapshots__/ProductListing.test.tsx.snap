// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductListing renders as expected 1`] = `
<div>
  <section
    class="product-list"
  >
    <a
      href="/app/productDetails/1"
    >
      <h3>
        Belt
      </h3>
      <img
        alt="Belt"
        class="product-image"
        src="https://live.staticflickr.com/7376/16529891306_3fed99f91c_b.jpg"
      />
      <p>
        £12
      </p>
    </a>
    <a
      href="/app/productDetails/2"
    >
      <h3>
        Jacket
      </h3>
      <img
        alt="Jacket"
        class="product-image"
        src="https://c2.staticflickr.com/8/7769/17963181310_29566ea829_b.jpg"
      />
      <p>
        £100
      </p>
    </a>
    <a
      href="/app/productDetails/3"
    >
      <h3>
        Shoes
      </h3>
      <img
        alt="Shoes"
        class="product-image"
        src="https://live.staticflickr.com/7671/17278913184_95385f21c7_b.jpg"
      />
      <p>
        £75
      </p>
    </a>
  </section>
</div>
`;
